esp-hal/src/spi/master.rs

```diff
@@ -713,31 +754,39 @@ where
     /// Enables both input and output functionality for the pin, and connects it
     /// to the MISO signal and SIO1 input signal.
     ///
+    /// Calling this will replace previous pin assignments for the SIO1 signal.
+    ///
     /// Use this if any of the devices on the bus use half-duplex SPI.
     ///
     /// The pin is configured to open-drain mode.
     ///
     /// Note: You do not need to call [Self::with_miso] when this is used.
     #[instability::unstable]
-    pub fn with_sio1<SIO1: PeripheralOutput>(self, miso: impl Peripheral<P = SIO1> + 'd) -> Self {
+    pub fn with_sio1<SIO1: PeripheralOutput>(
+        mut self,
+        miso: impl Peripheral<P = SIO1> + 'd,
+    ) -> Self {
         crate::into_mapped_ref!(miso);
         miso.enable_input(true);
         miso.enable_output(true);
 
         self.driver().info.miso.connect_to(&mut miso);
-        self.driver().info.sio1_output.connect_to(&mut miso);
+        // self.driver().info.sio1_output.connect_to(&mut miso);
+        self.sio1_pin = OutputConnection::connect_with_guard(miso, self.driver().info.sio1_output);
 
         self
     }
 
     /// Assign the SCK (Serial Clock) pin for the SPI instance.
     ///
+    /// Calling this will replace previous pin assignments for this signal.
```

Comment:
I would mention this last, or at least below the actually description of what the method does on all of these